README.md

```diff
@@ -7,6 +7,43 @@ Carbon is a tool for loading data into [Symplectic Elements](https://support.sym
 
 Please refer to the [mitlib-tf-workloads-carbon](https://github.com/mitlibraries/mitlib-tf-workloads-carbon) for the deployment configuration.
 
+For more information on the Carbon application, please refer to our [internal documentation on Confluence](https://mitlibraries.atlassian.net/l/cp/1E6cMvuT).
+
+## Data flow
+
+This flowchart depicts the data flow from MIT's Data Warehouse to the application's in-memory buffered streams and to the final location of the output file, an XML file on the Elements FTP server.
+
+```mermaid
+---
+title:
+---
+
+flowchart TB
+   subgraph ext-source[Database]
+      mit-dwrhs[(MIT Data Warehouse)]
+   end
+   
+   subgraph in-memory[Application In-memory]
+      direction TB
+      rec-generator([Record generator])
+      buffered-writer([Buffered Writer])
+      buffered-reader([Buffered Reader])
+   end
+
+   subgraph elements-ftp[Elements FTP server]
+      direction TB
+      xml-file([Feed XML file])
+   end
+
+   mit-dwrhs -->|Fetch query results | rec-generator
+   rec-generator-->|Yielding records one at a time, <br> transform record into normalized XML strings <br> and write to a file| buffered-writer
```

Comment:
What about changing:

> and write to a file

to:

> and passes to write buffer

Please pushback though if I'm misunderstanding how it works.

Reply:
I went back and forth on saying "write to file" (as in "file-like object") or "write to stream", but I do think "pass to write buffer" is clean and easy to understand.